src/components/page-donate/page-donate.tsx

```diff
@@ -19,54 +8,50 @@ export class PageDonate {
   @State() private amount?: number | null;
   @State() private showConfirmation: boolean = false;
   @State() private canNativeShare: boolean = false;
+  @State() private referral?: string | null;
 
   public componentWillLoad() {
     document.title = `Donate | Pizza to the Polls`;
-  }
+    const urlParams = new URLSearchParams(window.location.search);
+    this.referral = urlParams.get("referral") || "";
 
-  public render() {
-    let handler: any = null;
-    const StripeCheckout: any = (window as any).StripeCheckout;
+    if (urlParams.get("success") && urlParams.get("amount_usd")) {
+      this.amount = parseInt(urlParams.get("amount_usd") as string, 10);
```

Comment:
I'm surprised the `as string` is necessary. The type of `urlParams.get(…)` should be `string | undefined` and TypeScript can tell that we're in an `if` block that excludes the `undefined` case.

Certainly not a problem though.

Reply:
It doesn't seem like the linter (or compiler?) is smart enough for that--at least VSCode shows a red squiggly when I don't explicitly specify.